### ISSUE TITLE
[QUERY] How to take only one entity using TableClient.Query(filter, maxPerPage, select, cancellationToken)

### ISSUE THREAD

**Issue by haozhangms** [Azure/azure-sdk-for-net#26060]
### Library name and version

Azure.Data.Tables 12.3.0

### Query/Question

I'm working on migrating some code flows about azure storage table from WindowsAzure.Storage.Table to Azure.Data.Tables.
When using WindowAzure.Storage.Table, we can call TableQuery.Take(1) to get only one entity, just as following:
```
query = query.Take(1);
```
However, TableQuery class does not exist in newer version library, and we can only use TableClient.CreateQueryFilter() to construct certain query statement, which is used as the first parameter in TableClient.Query(). 

So could you please help me how to specify the count of retrieved result? Thanks!

### Environment

Microsoft Visual Studio Enterprise 2022 (64-bit) 
Version 17.0.1

**Comment by jsquire** [comment]
Hi @haozhangms.  Thank you for reaching out and we regret that you’re experiencing difficulties.  To return a single result, you’ll want to use `GetEntity` as discussed [here](https://github.com/Azure/azure-sdk-for-net/issues/25448).

**Comment by haozhangms** [comment]
Thanks for your comment, but it seems not be my desired solution. GetEntity() method requires two necessary parameters: PartitionKey and RowKey, however, I'm afraid I can't provide in some scenarios. 
For example, if I just want to know whether the storage table is empty or not, how to call GetEntity() method? or do you have any better solution? pls reply me in your convenience, thanks!

**Comment by ordinaryorange** [comment]
Just hit this issue too.
[The migration guide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/tables/Azure.Data.Tables/MigrationGuide.md) indicates this lib is an improvement/makes it easier to consume for developers
I'd argue the lack of `$Top()` support is a regression.
`GetEntity` is inadequate as a substitute, there are many situations where a filter predicate is needed on the call, but only the `$Top=N` desired on the response.
The proposed workaround in #25448 is to use pagination, which results in the following .Net client code

`tableClient.Query<T>(filter, 1).First()`

if you forget that the `Query` returns a `Azure.Pageable` and just consume the `IEnumerable<T>` then the following will return > 1 entities

`tableClient.Query<T>(filter, 1).ToList()`

IMHO that is not more intuitive to a developer

**Comment by christothes** [comment]
The rationale for not allowing the IQueryable extension methods on the Query method is that it can encourage inefficient queries while hiding the fact that they are expensive.  For example, in a query like below, the syntax is simpler but it hides the fact that the service call is, in fact, returning the maximum number of results and filtering the first item on the client side.

```c#
tableClient.Query<T>(filter, 1).ToList().First()
```

Note also that the $top query parameter is more accurately named in the new client as `maxPerPage`. The service only takes this as a hint for how many results to return per page, but will always return the full result set if all pages are consumed. This is behavior of the service that is agnostic to which SDK client is used.  

In summary, the new client does intentionally require a bit more complexity to query for a limited number of results, but this is designed to avoid unintentionally expensive queries.
